src/LinkDotNet.NCronJob/NCronJobExtensions.cs

```diff
@@ -27,11 +30,13 @@ public static class NCronJobExtensions
     {
         // 4 is just an arbitrary multiplier based on system observed I/O, this could come from Configuration
         var settings = new ConcurrencySettings { MaxDegreeOfParallelism = Environment.ProcessorCount * 4 };
-        services.AddSingleton(settings);
-
         var builder = new NCronJobOptionBuilder(services, settings);
         options?.Invoke(builder);
 
+        if (services.IsServiceRegistered<CronScheduler>())
```

Comment:
Currently, I have two ways of thinking:
1. If we use this approach, this should be one line 1 so to speak, not?
2. Or alternatively: Wouldn't it be easier to make that function idempotent?

Basically, always calling `TryAddSingleton` etc (`AddHostedService` already uses `TryAddSingleton` under the hood).

Reply:
As far as I remember I don't believe calling AddHostedService is idempotent. I'll verify this

Reply:
By the way, I added the idempotent version. `AddHostedService` does indeed call `TryAddEnumerable` which will not work if the same instance is already registered.